### DISCUSSION TITLE
Uncaught 500 error from API route behavior?

### DISCUSSION THREAD

**Discussion by bscaspar** [vercel/next.js#15640] [Help] [answered]
I don't quite understand how errors (other than 404) are handled. What is the expected behavior if an API route returns an uncaught 500 error? Should the default 500 error page be displayed? If not, what is an example of when the 500 error page should be displayed?

**Comment by jamesmosier** [comment, marked as answer]
500 errors for an API route would not display an HTML page, rather just a server HTTP status code of 500. 

As far as *when* the 500 error page would show, one example I can think of is if you have a page that uses `getServerSideProps` and that has an error in it.

```js
export async function getServerSideProps(context) {
  throw new Error("500 page will be shown")
}
```

Basically any runtime error that happens on a page while running on the server (aka Node) would cause this to happen.

**Reply by bscaspar** [reply]
Makes sense, thank you!
